pypeit/pypeit.py

```diff
@@ -74,47 +74,51 @@ def __init__(self, pypeit_file, verbosity=2, overwrite=True, reuse_masters=False
                  show=False, redux_path=None):
 
         # Load
-        cfg_lines, data_files, frametype, usrdata, setups = parse_pypeit_file(pypeit_file, runtime=True)
+        cfg_lines, data_files, frametype, usrdata, setups \
+                = parse_pypeit_file(pypeit_file, runtime=True)
         self.pypeit_file = pypeit_file
 
         # Spectrograph
         cfg = ConfigObj(cfg_lines)
         spectrograph_name = cfg['rdx']['spectrograph']
         self.spectrograph = load_spectrograph(spectrograph_name)
 
-        # Par
-        # Defaults
-        spectrograph_def_par = self.spectrograph.default_pypeit_par()
-        # Grab a science file for configuration specific parameters
+        # --------------------------------------------------------------
+        # Get the full set of PypeIt parameters
+        #   - Grab a science file for configuration specific parameters
         for idx, row in enumerate(usrdata):
             if 'science' in row['frametype']:
                 sci_file = data_files[idx]
                 break
-        # Set
-        spectrograph_cfg_lines = self.spectrograph.config_specific_par(spectrograph_def_par, sci_file).to_config()
+        #   - Configuration specific parameters for the spectrograph
+        spectrograph_cfg_lines = self.spectrograph.config_specific_par(sci_file).to_config()
+        #   - Build the full set, merging with any user-provided
+        #     parameters
         self.par = PypeItPar.from_cfg_lines(cfg_lines=spectrograph_cfg_lines, merge_with=cfg_lines)
+        # --------------------------------------------------------------
 
-        # Fitstbl
-        self.fitstbl = PypeItMetaData(self.spectrograph, self.par, file_list=data_files,
+        
+        # --------------------------------------------------------------
+        # Build the meta data
+        #   - Re-initilize based on the file data
+        self.fitstbl = PypeItMetaData(self.spectrograph, self.par, files=data_files,
                                       usrdata=usrdata, strict=True)
-        # The following could be put in a prepare_to_run() method in PypeItMetaData
-        if 'setup' not in self.fitstbl.keys():
-            self.fitstbl['setup'] = setups[0]
-        self.fitstbl.get_frame_types(user=frametype)  # This sets them using the user inputs
-        self.fitstbl.set_defaults()  # Only does something if values not set in PypeIt file
-        self.fitstbl._set_calib_group_bits()
-        self.fitstbl._check_calib_groups()
-        # Write .calib file (For QA naming amongst other things)
+        #   - Interpret automated or user-provided data from the PypeIt
+        #   file
+        self.fitstbl.finalize_usr_build(frametype, setups[0])
```

Comment:
Thanks for this.

Reply:
NP.  Hopefully the doc string for `finalize_usr_build` is clear as to what it's doing.